addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -2264,7 +2264,11 @@ export class PosStore extends Reactive {
 
     selectEmptyOrder() {
         const emptyOrders = this.models["pos.order"].filter(
-            (order) => order.is_empty() && !order.finalized
+            (order) =>
+                order.is_empty() &&
+                !order.finalized &&
+                order.payment_ids.length === 0 &&
+                !order.partner_id
```

Comment:
```suggestion
                order.is_empty() &&
                !order.finalized &&
                order.payment_ids.length === 0 &&
                !order.partner_id &&
                order.pricelist_id?.id === this.config.pricelist_id?.id &&
                order.fiscal_position_id?.id === this.config.default_fiscal_position_id?.id
```

Don't forget to check pricelist and fiscal position :)

Reply:
Can you also test this scenario ?